.github/workflows/test_e2eshark.yml

```diff
@@ -0,0 +1,100 @@
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+name: E2ESHARK Test Suite
+on:
+  workflow_dispatch:
+  schedule:
+  # Runs at 1:00 AM UTC, which is 5:00 AM PST (UTC-8)
+   - cron: '0 13 * * *'
+
+jobs:
+  turbine:
+    runs-on: nodai-amdgpu-mi250-x86-64
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+        with:
+          repository: nod-ai/SHARK-Turbine
+          ref: torch_2.1_support
+          path: turbine
+
+  e2eshark:
+    needs: [turbine]
+    runs-on: nodai-amdgpu-mi250-x86-64
+    env:
+      E2E_VENV_DIR: ${{ github.workspace }}/test-suite_venv
+      TURBINE_VENV_DIR: ${{ github.workspace }}/turbine_venv
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+        with:
+          repository: nod-ai/SHARK-TestSuite
+          path: test-suite
+
+      - name: "Setup e2eshark python venv"
+        run: |
+          python3.11 -m venv ${E2E_VENV_DIR}
+          pip install --upgrade pip
+          pip install -r ./e2eshark/requirements.txt
+        working-directory: ./test-suite
+
+      - name: Run Onnx Mode
+        run: |
+          source ${E2E_VENV_DIR}/bin/activate
+          cd e2eshark
+          free -mh
+          HF_TOKEN=${{ secrets.HF_TOKEN }} AZ_PRIVATE_CONNECTION=${{ secrets.AZ_PRIVATE_CONNECTION }} python3.11 ./run.py -r ./test-onnx --report --cachedir ~/huggingface_cache --mode onnx -f onnx pytorch -g models --tolerance .001 .001 --cleanup --postprocess --ci -v
+        working-directory: ./test-suite
+
+      - name: Steup turbine python venv
+        run: |
+          python3.11 -m venv ${TURBINE_VENV_DIR}
+          source ${TURBINE_VENV_DIR}/bin/activate
+          pip install --upgrade pip
+          pip install --index-url https://download.pytorch.org/whl/cpu \
+            -r ../turbine/core/pytorch-cpu-requirements.txt \
+            -r ../turbine/core/torchvision-requirements.txt
+          pip install --upgrade -r ../turbine/core/requirements.txt
+          pip install -e ../turbine/core[testing]
+          pip install -e ../turbine/models
+          pip install -r ./e2eshark/turbine-requirements.txt
+          pip uninstall -y transformers
+          pip install transformers -U
+        working-directory: ./test-suite
+
+      - name: Run Turbine Mode
+        run: |
+          source ${TURBINE_VENV_DIR}/bin/activate
+          cd e2eshark
+          free -mh
+          HF_TOKEN=${{ secrets.HF_TOKEN }} python3.11 ./run.py -r ./test-turbine --report --cachedir ~/huggingface_cache --mode turbine -g models --cleanup --postprocess --ci -v
+        working-directory: ./test-suite
+
+  upload_artifacts:
+    needs: [e2eshark]
+    runs-on: nodai-amdgpu-mi250-x86-64
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+        with:
+          repository: nod-ai/e2eshark-reports
+          ref: 'main'
+          token: ${{ secrets.E2ESHARK_GITHUB_TOKEN }}
+          path: e2eshark-reports
```

Comment:
It looks like this is depending on running on the same machine with some files persisting. If there are multiple runners matching this label then that will not always be true. If there are multiple actions running then the files may be overwritten between when the previous job runs and when this job runs. Instead, you could either
* put all the jobs together
* pass report files between jobs: https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts#passing-data-between-jobs-in-a-workflow

Reply:
Cool, this is good to know (will be helpful with more intense jobs that need to pass artifacts). Here, I will just combine them because the upload is really quick and can just be combined.

Reply:
You can also pass files between jobs using cloud storage (especially if they are too large for GitHub Artifacts).
We do that in IREE today: 
* https://github.com/iree-org/iree/blob/e4b5a9350aee7cfafefb440a7f01b3d376842d92/.github/workflows/benchmark_execution.yml#L192-L193
* https://github.com/iree-org/iree/blob/e4b5a9350aee7cfafefb440a7f01b3d376842d92/.github/workflows/ci.yml#L986-L1001

(but I've wanted to move some of those to workflow artifacts using standard github infra if possible)

Reply:
It will also help when running across multiple machines (e.g. w7900, MI250, MI300). Distribute tests across workers -> wait on all workers and aggregate the results

Reply:
Made changes for data passing between jobs